src/core/opamSystem.ml

```diff
@@ -129,19 +129,15 @@ let win32_unlink fn =
     with _ -> raise e)
 
 let remove_file_t ?(with_log=true) file =
-  if
-    try ignore (Unix.lstat file); true with Unix.Unix_error _ -> false
```

Comment:
`remove_file_t` on a non-existing file used to simply ignore it. This PR changes this behaviour to error instead.
Was the old behaviour used anywhere that we know?

Reply:
I don't think so - it's only other use is in the cross-device fallback for `mv`. The only way it could be called with a file which didn't exist is if the file has been removed between the test in the calling function and the test here, and there was already a race for that anyway (between the test I've deleted here and the `internal_error` handler below).

Reply:
it is also being used in `remove_file` and `remove_dir` which are used everywhere in opam's code (e.g. via `OpamFilename.remove`)

Reply:
`remove_file` doesn't use `remove_file_t` (LL202-215) and `remove_dir` already checks whether its argument exists?

Reply:
AH! I see. There are two toplevel versions of `remove_file` in the same file. Only the latter is exported. Sorry for missing that. This is all good then.

Reply:
Actually, until you’d said, I’d missed that too! I was only sure that the code was ok because I’d deleted the first remove_file as a result of an unused value warning and already searched for other uses of remove_file_t 🙂